### PR TITLE
[gui-tests][full-ci] retry failed scenario

### PR DESCRIPTION
**Pull request by saw-jan**
Implements retry failing test scenario.

### Condition to trigger retry:
- `full-ci` in the PR titile
- tag and nightly builds

>[!NOTE]
>`--retry` and `--abortOnFail` options clash with each other.
>So, we enable `--abortOnFail` on normal PR
>and enable `--retry` if above conditions are met

### What happens when retry is triggered?
- failed test scenario is retried once
- on retry run video recording is enabled

### Other changes:
- renamed env `SCREEN_RECORD_ON_FAILURE` to `RECORD_VIDEO_ON_FAILURE`
- renamed config `screenRecordOnFailure` to `recordVideoOnFailure`
- added config `videoRecordLimit` to limit video recordings of failed scenarios - `default: 5` (because of large size)
- moved reports related functions to separate `ReportHelper` file

### CODE REVIEW

File: test/gui/shared/scripts/helpers/ConfigHelper.py [hunk 1]
```diff
@@ -73,12 +73,14 @@ def get_default_home_dir():
     'clientConfigDir': 'CLIENT_CONFIG_DIR',
     'guiTestReportDir': 'GUI_TEST_REPORT_DIR',
     'ocis': 'OCIS',
-    'screenRecordOnFailure': 'SCREEN_RECORD_ON_FAILURE',
+    'recordVideoOnFailure': 'RECORD_VIDEO_ON_FAILURE',
```
Comment (ishabaral) [inline test/gui/shared/scripts/helpers/ConfigHelper.py:76]: how about writing these attributes in snake_case?

File: test/gui/shared/scripts/helpers/ConfigHelper.py [hunk 2]
```diff
@@ -73,12 +73,14 @@ def get_default_home_dir():
...
 }
 
 DEFAULT_PATH_CONFIG = {
     'custom_lib': os.path.abspath('../shared/scripts/custom_lib'),
     'home_dir': get_default_home_dir(),
+    # allow to record first 5 videos
+    'videoRecordLimit': 5,
```
Comment (ishabaral) [inline test/gui/shared/scripts/helpers/ConfigHelper.py:83]:
```suggestion
    'video_record_limit': 5,
```

File: test/gui/shared/scripts/helpers/ConfigHelper.py (hunk 1 above)
Comment (saw-jan) [reply test/gui/shared/scripts/helpers/ConfigHelper.py:76]: would be great. but not in this PR.

Comment (saw-jan) [reply test/gui/shared/scripts/helpers/ConfigHelper.py]: Remaining attributes can be renamed in the future PRs.

**Review by ishabaral — APPROVED** [review]

**Pull request merged**